README.md
[![Actions Status](https://github.com/kaz-utashiro/greple-subst-desumasu/workflows/test/badge.svg)](https://github.com/kaz-utashiro/greple-subst-desumasu/actions) [![MetaCPAN Release](https://badge.fury.io/pl/App-Greple-subst-desumasu.svg)](https://metacpan.org/release/App-Greple-subst-desumasu)
# NAME

App::Greple::subst::desumasu - Japanese DESU/MASU dictionary for App::Greple::subst

# SYNOPSIS

    greple -Msubst::desumasu --dearu --subst --all file

    greple -Msubst::desumasu --dearu --diff file

    greple -Msubst::desumasu --dearu --replace file

# DESCRIPTION

greple -Msubst module based on
[desumasu-converter](https://github.com/kssfilo/desumasu-converter).

This is a simple checker/converter module for the Japanese writing
styles so called DESU/MASU (ですます調: 敬体) and DEARU (である調: 常
体).  This is not my own idea and the dictionary is based on
[https://github.com/kssfilo/desumasu-converter](https://github.com/kssfilo/desumasu-converter).

See article [https://kanasys.com/tech/722](https://kanasys.com/tech/722) for detail.

# OPTIONS

- **--dearu**
- **--dearu-n**
- **--dearu-N**

    Convert DESU/MASU to DEARU style.

    DESU (です) and MASU (ます) are sometimes followed by NE (ね) in an
    open situation, and that NE (ね) is removed from the converted result
    by default.  Option with **-n** keeps the NE (ね), and option with
    **-N** igonores it.

- **--desumasu**
- **--desumasu-n**
- **--desumasu-N**

    Convert DEARU to DESU/MASU style.

Use them with **greple** **-Msubst** options.

- **--subst --all --no-color**

    Print converted text.

- **--diff**

    Produce diff output of original and converted text.  Use **cdif**
    command in [App::sdif](https://metacpan.org/pod/App%3A%3Asdif) to visualize the difference.

- **--create**
- **--replace**
- **--overwrite**

    To update the file, use these options.  Option **--create** make new
    file with extention `.new`.  Option **--replace** updates the target
    file with backup, while option **--overwrite** does it without backup.

See [App::Greple::subst](https://metacpan.org/pod/App%3A%3AGreple%3A%3Asubst) for other options.

# INSTALL

## CPANMINUS

From CPAN:

    cpanm App::Greple::subst::desumasu

From GIT repository:

    cpanm https://github.com/kaz-utashiro/greple-subst-desumasu.git

# SEE ALSO

[App::Greple](https://metacpan.org/pod/App%3A%3AGreple), [App::Greple::subst](https://metacpan.org/pod/App%3A%3AGreple%3A%3Asubst)

[App::sdif](https://metacpan.org/pod/App%3A%3Asdif)

[https://github.com/kssfilo/desumasu-converter](https://github.com/kssfilo/desumasu-converter),
[https://kanasys.com/tech/722](https://kanasys.com/tech/722)

[greple で「ですます調」を「である化」する](https://qiita.com/kaz-utashiro/items/8f4878300043ce7b73e7)

# AUTHOR

Kazumasa Utashiro

# LICENSE

Copyright 2021-2023 Kazumasa Utashiro.

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.


Changes
Revision history for Perl extension App-Greple-subst-desumasu

{{$NEXT}}

0.07 2023-04-03T01:30:19Z

    - 「はない」→「はありません」を辞書に追加
    - ドキュメント中のタイポを修正

0.06 2023-02-25T01:31:51Z

    - 否定形の表現を追加

0.05 2022-10-09T02:25:44Z

    - require File::Share@0.27

0.04 2022-07-08T03:28:03Z

    - 「です -> だ」より「です -> である」を優先

0.03 2022-07-08T01:06:57Z

    - 「である -> です」ルールを追加

0.02 2022-04-29T07:11:30Z

    - update test and workflow

0.01 2022-04-22T00:33:42Z

    - original version



lib/App/Greple/subst/desumasu.pm
package App::Greple::subst::desumasu;

our $VERSION = "0.07";

use 5.014;
use warnings;
use utf8;

use App::Greple::subst;

use File::Share qw(:all);
$ENV{DESUMASU_DICT} = dist_dir __PACKAGE__ =~ s/::/-/gr;

1;

=encoding utf-8

=head1 NAME

App::Greple::subst::desumasu - Japanese DESU/MASU dictionary for App::Greple::subst

=head1 SYNOPSIS

    greple -Msubst::desumasu --dearu --subst --all file

    greple -Msubst::desumasu --dearu --diff file

    greple -Msubst::desumasu --dearu --replace file

=head1 DESCRIPTION

greple -Msubst module based on
L<desumasu-converter|https://github.com/kssfilo/desumasu-converter>.

This is a simple checker/converter module for the Japanese writing
styles so called DESU/MASU (ですます調: 敬体) and DEARU (である調: 常
体).  This is not my own idea and the dictionary is based on
L<https://github.com/kssfilo/desumasu-converter>.

See article L<https://kanasys.com/tech/722> for detail.

=head1 OPTIONS

=over 7

=item B<--dearu>

=item B<--dearu-n>

=item B<--dearu-N>

Convert DESU/MASU to DEARU style.

DESU (です) and MASU (ます) are sometimes followed by NE (ね) in an
open situation, and that NE (ね) is removed from the converted result
by default.  Option with B<-n> keeps the NE (ね), and option with
B<-N> igonores it.

=item B<--desumasu>

=item B<--desumasu-n>

=item B<--desumasu-N>

Convert DEARU to DESU/MASU style.

=back

Use them with B<greple> B<-Msubst> options.

=over 7

=item B<--subst --all --no-color>

Print converted text.

=item B<--diff>

Produce diff output of original and converted text.  Use B<cdif>
command in L<App::sdif> to visualize the difference.

=item B<--create>

=item B<--replace>

=item B<--overwrite>

To update the file, use these options.  Option B<--create> make new
file with extention C<.new>.  Option B<--replace> updates the target
file with backup, while option B<--overwrite> does it without backup.

=back

See L<App::Greple::subst> for other options.

=head1 INSTALL

=head2 CPANMINUS

From CPAN:

    cpanm App::Greple::subst::desumasu

From GIT repository:

    cpanm https://github.com/kaz-utashiro/greple-subst-desumasu.git

=head1 SEE ALSO

L<App::Greple>, L<App::Greple::subst>

L<App::sdif>

L<https://github.com/kssfilo/desumasu-converter>,
L<https://kanasys.com/tech/722>

L<greple で「ですます調」を「である化」する|https://qiita.com/kaz-utashiro/items/8f4878300043ce7b73e7>

=head1 AUTHOR

Kazumasa Utashiro

=head1 LICENSE

Copyright 2021-2023 Kazumasa Utashiro.

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut

__DATA__

option default -Msubst

define DICTDIR $ENV{DESUMASU_DICT}

option --desumasu --dict DICTDIR/desumasu.dict
option --dearu    --dict DICTDIR/dearu.dict

option --desumasu-n --dict DICTDIR/desumasu-keep-ne.dict
option --dearu-n    --dict DICTDIR/dearu-keep-ne.dict

option --desumasu-N --dict DICTDIR/desumasu-ignore-ne.dict
option --dearu-N    --dict DICTDIR/dearu-ignore-ne.dict

#  LocalWords:  diff cdif
